projects/plugins/automattic-for-agencies-client/src/js/components/connected-card/index.jsx

```diff
@@ -0,0 +1,214 @@
+import restApi from '@automattic/jetpack-api';
+import { Button, Spinner } from '@automattic/jetpack-components';
+import { CONNECTION_STORE_ID } from '@automattic/jetpack-connection';
+import { Modal } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import classNames from 'classnames';
+import React, { useCallback, useEffect, useState } from 'react';
+import BrandedCard from '../branded-card';
+import CheckCircleIcon from '../check-circle-icon';
+import CloseCircleIcon from '../close-circle-icon';
+import styles from './styles.module.scss';
+
+/**
+ * Site Connected Content component.
+ *
+ * @returns {React.Component} The `SiteConnectedContent` component.
+ */
+function SiteConnectedContent() {
+	const navigateToDashboard = useCallback( () => {
+		window.location.href = 'https://agencies.automattic.com';
+	}, [] );
+
+	return (
+		<div className={ styles.card }>
+			<h1>
+				{ __(
+					'Your site is connected to Automattic for Agencies',
+					'automattic-for-agencies-client'
+				) }
+			</h1>
+			<p
+				className={ classNames(
+					styles.connection_status,
+					styles[ 'connection_status--connected' ]
+				) }
+			>
+				<CheckCircleIcon />
+				{ __( 'Site is connected and syncing', 'automattic-for-agencies-client' ) }
+			</p>
+			<div>
+				<Button onClick={ navigateToDashboard } isExternalLink>
+					{ __( 'Visit the dashboard', 'automattic-for-agencies-client' ) }
+				</Button>
+			</div>
+		</div>
+	);
+}
+
+/**
+ * Site Disconnected Content component.
+ *
+ * @returns {React.Component} The `SiteDisconnectedContent` component.
+ */
+function SiteDisconnectedContent() {
+	const { setConnectionStatus } = useDispatch( CONNECTION_STORE_ID );
+
+	const refreshConnectionState = useCallback( () => {
+		setConnectionStatus( { isActive: false, isRegistered: false, isUserConnected: false } );
+	}, [ setConnectionStatus ] );
```

Comment:
Essentially, waiting until the user clicks "reconnect" to update the redux state with the newly disconnected state. This will take them back to the connection flow.